packages/signers/src/__tests__/signable-message-test.ts

```diff
@@ -0,0 +1,42 @@
+import { TextEncoder } from 'text-encoding-impl';
+
+import { createSignableMessage } from '../signable-message';
+
+describe('createSignableMessage', () => {
+    it('creates a SignableMessage from a byte array', () => {
+        // Given a byte array.
+        const content = new Uint8Array([1, 2, 3]);
+
+        // When we use it to create a new SignableMessage.
+        const message = createSignableMessage(content);
+
+        // Then we expect a message with the same content and no signatures.
+        expect(message.content).toBe(content);
+        expect(message.signatures).toStrictEqual({});
+    });
+
+    it('creates a SignableMessage with signatures', () => {
+        // Given a byte array content and some signatures.
+        const content = new Uint8Array([1, 2, 3]);
+        const signatures = {
+            '1111': new Uint8Array([1, 1, 1, 1]),
+            '2222': new Uint8Array([2, 2, 2, 2]),
+        };
+
+        // When we create a SignableMessage using both the content and the existing signatures.
+        const message = createSignableMessage(content, signatures);
+
+        // Then we expect the message to contain both of these argument as-is.
+        expect(message.content).toBe(content);
+        expect(message.signatures).toBe(signatures);
+    });
+
+    it('creates a SignableMessage from a UTF-8 string', () => {
+        // When we create a SignableMessage by providing a string.
+        const message = createSignableMessage('Hello world!');
+
+        // Then we expect this string to be UTF-8 encoded.
+        expect(message.content).toStrictEqual(new TextEncoder().encode('Hello world!'));
+        expect(message.signatures).toStrictEqual({});
+    });
+});
```

Comment:
I had the same thoughts. Initially I was planning on bringing `codecs-strings` but then thought let's at least narrow it to the TextEncoder impl package.

I'd say 99% of the time, when people sign messages, it's from a UTF-8 string so it would be a nice helper.

**I'm happy to make it only accept `Uint8Array`** but it kinda broke my heart when this piece of the documentation (the very first code snippet of the README) when from this:

```ts
const myMessage = createSignableMessage('Hello world!');
const [messageSignatures] = await mySigner.signMessage([myMessage]);
```

to this:

```ts
const myMessage = createSignableMessage(new TextEncoder().encode('Hello world!'));
const [messageSignatures] = await mySigner.signMessage([myMessage]);

// or even:
const myMessage = createSignableMessage(getUtf8Encoder().encode('Hello world!'));
const [messageSignatures] = await mySigner.signMessage([myMessage]);
```

What do you think?

Reply:
I vote bringing `codecs-strings` to keep the code looking like the first snippet.

Reply:
I'm fine with that, it's just RIP now everyone has a peer dependency on `fastestsmallesttextencoderdecoder` for React Native. It won't get bundled, of course, but it will get downloaded.

Reply:
Right now, I'm not convince that `Record<Address, SignatureBytes>` is the best definition of a `SignatureDirectory` but I used it so it matches the definition of the `signatures` attribute of transactions.

I think `Record<Address, Signature>` may be more elegant since both the keys and values will be encoded as base 58 strings.

The reason I mention that here is that, if we make that change, we'll likely need `getBase58Encoder()` in this package anyway.